core/types/rollup_cost_test.go

```diff
@@ -503,3 +503,47 @@ func TestFlzCompressLen(t *testing.T) {
 		require.Equal(t, tc.expectedLen, output)
 	}
 }
+
+// copy of emptyTx with non-zero gas
+var emptyTxWithGas = NewTransaction(
+	0,
+	common.HexToAddress("095e7baea6a6c7c4c2dfeb977efac326af552d87"),
+	big.NewInt(0), bedrockGas.Uint64(), big.NewInt(0),
+	nil,
+)
+
+// TestTotalRollupCostFunc tests that the total rollup cost function correctly
+// combines the L1 cost and operator cost.
+func TestTotalRollupCostFunc(t *testing.T) {
+	zero := uint64(0)
+	later := uint64(10)
+	config := &params.ChainConfig{
+		Optimism:     params.OptimismTestConfig.Optimism,
+		RegolithTime: &zero,
+		EcotoneTime:  &zero,
+		FjordTime:    &zero,
+		HoloceneTime: &zero,
+		IsthmusTime:  &later,
+	}
+	statedb := &testStateGetter{
+		baseFee:             baseFee,
+		overhead:            overhead,
+		scalar:              scalar,
+		blobBaseFee:         blobBaseFee,
+		baseFeeScalar:       uint32(baseFeeScalar.Uint64()),
+		blobBaseFeeScalar:   uint32(blobBaseFeeScalar.Uint64()),
+		operatorFeeScalar:   uint32(operatorFeeScalar.Uint64()),
+		operatorFeeConstant: operatorFeeConstant.Uint64(),
+	}
+
+	costFunc := NewTotalRollupCostFunc(config, statedb)
+	cost := costFunc(emptyTxWithGas, later/2)
```

Comment:
Nit: why not choose `later - 1` and `later+1`? This might be kinder for any future modification of this test where a new fork modifies how fees are charged...

Reply:
I can do that, I just wanted something before and after so choose /2 and *2. But happy to change it to -+1. Although that author of the new test could then also do it if they need it.

Reply:
done